M/mlpack/build_tarballs.jl

```diff
@@ -6,15 +6,24 @@ using BinaryBuilder
 
 # Set sources and other environment variables.
 name = "mlpack"
-version = v"3.4.3"
-source_version = v"3.4.2"
+source_version = v"4.0.0"
+version = source_version
 sources = [
     ArchiveSource("https://www.mlpack.org/files/mlpack-$(source_version).tar.gz",
-                  "9e5c4af5c276c86a0dcc553289f6fe7b1b340d61c1e59844b53da0debedbb171"),
-    DirectorySource("./bundled"),
+                  "041d9eee96445667d2f7b970d2a799592027f1f8818cd96a65dcce1ac0745773"),
+    # STB headers for image support.
+    ArchiveSource("https://www.mlpack.org/files/stb.tar.gz",
+                  "5c01115d6909621af63a502437334f2014065a5ce6bb7ac05acbf15f24acce15"),
```

Comment:
Can this be a separate header-only package like cereal?  Also, is this tarball unversioned?  If it'll be replaced in the future this would make this recipe unreproducible.

Reply:
Yeah, good point.  STB is only two header files so I thought I could keep it simple, but let me just open another header-only package for it---it'll be quite trivial.

Reply:
Now handled via #5741.